.github/workflows/e2e-cad.yml

```diff
@@ -0,0 +1,55 @@
+name: End-to-end CAD flow smoketests
+
+on:
+  push:
+    branches:
+      - main
+      - test-ci
+      - sky130_pydantic
+  pull_request:
+
+jobs:
+  sky130-cm-e2e:
+    runs-on: [self-hosted] # TODO: replace with something centrally admined
+
+    steps:
+        # Clone this repo
+      - name: Checkout
+        uses: actions/checkout@v4.1.6
+
+      - name: Run syn (Genus), par (Innovus), drc (Pegasus), and lvs (Pegasus) on the GCD design in sky130, using sky130_scl.
+        run: |
+          cd e2e
+          poetry install
+          source $(poetry env info --path)/bin/activate
+          # redirect all hammer output a temp file, since it contains proprietary tool logs
+          tempfile=$(mktemp)
+
+          # run cadence tools on design
+          echo "log file is $tempfile on $(hostname)" | tee -a $tempfile
+          echo "End-to-end CAD flow CI smoketest running on" ${{ github.head_ref }}.${{ github.sha }} > $tempfile 2>&1
+          make build >> $tempfile 2>&1
+          echo "running par.." | tee -a $tempfile
+          make par >> $tempfile 2>&1
+          echo "running drc.." | tee -a $tempfile
+          make drc >> $tempfile 2>&1
+          echo "running lvs.." | tee -a $tempfile
+          make lvs >> $tempfile 2>&1
```

Comment:
I'm not a huge fan of dumping something out to a file if it's sensitive. What happens if a current Hammer user accidentally removes one of these echos on accident and info is leaked? Maybe it's better to try to disable actions from logging entirely?

Reply:
Hmm, I see your point but I don't see how that reasoning doesn't apply to any output masking/secret hiding in an action. I've added a warning comment to hopefully dissuade accidental leaking, I'm not really sure how to [actually resolve this concern](https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/about-self-hosted-runners#self-hosted-runner-security)- maybe only allow maintainers to manually kick off the workflow once they've checked if no malicious changes to the actions? That seems like a single point of failure though